src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -161,6 +252,64 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string paramete
             }
         }
 
+        /// <summary>
+        /// Extracts a nested template in the exact format as user input, accounting for all white space and formatting
+        /// </summary>
+        /// <param name="template">Parent template containing nested template</param>
+        /// <param name="startOfTemplate">Line number where the nested template starts</param>
+        /// <returns>A nested template string</returns>
+        private string ExtractNestedTemplate(string template, int startOfTemplate)
+        {
+            bool startOfNestingFound = false;
+            int lineNumberCounter = 1;
+            int curlyBraceCounter = 0;
+            string stringNestedTemplate = "";
+            foreach (var myString in template.Split(Environment.NewLine))
+            {
+                if (lineNumberCounter < startOfTemplate)
+                {
+                    lineNumberCounter += 1;
+                    continue;
+                }
+                if (!startOfNestingFound)
+                {
+                    if (myString.Contains('{'))
+                    {
+                        stringNestedTemplate += myString.Substring(myString.IndexOf('{'));
+                        stringNestedTemplate += Environment.NewLine;
+                        startOfNestingFound = true;
+                        lineNumberCounter += 1;
+                        curlyBraceCounter += 1;
+                    }
+                    continue;
+                }
+                // after finding the start of nesting, count the opening and closing braces till they match up
+                int inlineCounter = 1;
+                foreach (char c in myString)
+                {
+                    if (c == '{') curlyBraceCounter++;
+                    if (c == '}') curlyBraceCounter--;
+                    if (curlyBraceCounter == 0) // done
+                    {
+                        stringNestedTemplate += myString[..inlineCounter];
+                        break;
```

Comment:
you can also `return` here instead and then remove the ` if (curlyBraceCounter == 0)
                {
                    break;
                }` section